webpack/scenes/ContentViews/Table/tableDataGenerator.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { compoundExpand } from '@patternfly/react-table';
+import { ScreenIcon, RepositoryIcon, ContainerNodeIcon } from '@patternfly/react-icons';
+import { translate as __ } from 'foremanReact/common/I18n';
+
+import IconWithCount from '../components/IconWithCount';
+import DetailsExpansion from '../expansions/DetailsExpansion';
+import RepositoriesExpansion from '../expansions/RepositoriesExpansion';
+import EnvironmentsExpansion from '../expansions/EnvironmentsExpansion';
+import VersionsExpansion from '../expansions/VersionsExpansion';
+import ContentViewName from '../components/ContentViewName';
+
+export const buildColumns = () => [
+  __('Name'), __('Last published'), __('Details'),
+  { title: __('Environments'), cellTransforms: [compoundExpand] },
+  { title: __('Repositories'), cellTransforms: [compoundExpand] },
+  { title: __('Versions'), cellTransforms: [compoundExpand] },
+];
+
+const buildRow = (contentView, openColumn) => {
+  const {
+    id, composite, name, environments, repositories, versions, last_published: lastPublished,
+  } = contentView;
+  const row = [
+    { title: <ContentViewName composite={composite ? 1 : undefined} name={name} cvId={id} /> },
+    lastPublished || 'Not yet published',
+    { title: __('Details'), props: { isOpen: false, ariaControls: `cv-details-expansion-${id}`, contentviewid: id } },
```

Comment:
After thinking a bit more over lunch, maybe it's better to handle the data this way:

Your formatted data will be not just the array of rows, but an array of objects like
```js
{
isExpanded: false,
contentViewId: 1,
contentViewDetails: {..},
rowData: [...],
}
```
That way you have all the data where you need it, attached to the row data.  And then you can map THAT to get your `rows` prop to pass down.

Reply:
(This does maybe put the `isExpanded` state in Redux where it doesn't need to be.  But maybe you could map that separately in the component state.)

Reply:
Thanks, let me play around with it. I'm leaning towards some sort of row mapping as well, but not sure how much should live in redux and how much should be in the component. I'll see what feels right!